woocommerce/woocommerce-gateway-amazon-pay
# Release branch/2.5.2

### All Submissions:






### Changes proposed in this Pull Request:



Closes:
 
#252, #257, #256, #263, #207

### How to test the changes in this Pull Request:

1. Install a build of this version 
2. Set up Amazon standard account
for #252 
   3. Set 4 decimals for EUR or USD
   4. Go to the site and add the product to the cart, 
   5. go to the cart and open the console of the browser, 
   6. no related console error should be displayed, the decimals should be truncated to the default for each currency.
for #257 
   7. Setup Amazon with Price based on country
   8. Add product to cart and use the Express Checkout
   9. Select other shipping address that changes the amount (like JPY)
   10. Try to place order.
   11. It should remove the decimals.
for #256 
   12. when selecting "Enable hide button mode "- OR -" string should be hidden on minicart
for #263 
    13. Purchase a subscription with PayPal or Stripe.
    14. Go to Subscription and click on "Change Payment Method"
    15. Select "Amazon" or click on the Amazon Pay Express button, 
    16. Payment Method changed without a problem.
for #207:
    17. Install and activate WooCommerce Product Add-Ons or Gravity Forms Product Add-ons
    18. Set up some variables that add some value to the product
    19. With the user go to the product page and select the variable that adds value to the price
    20. Click on the Express checkout of Amazon Pay and log in to the account
    21. The thank you page doesn't show the variable that was added by user

### Other information:




### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

* Fix - Error when using a non Standard decimal amount for a currency.
* Fix - chargeAmount.Amount is invalid when changing currency after selecting other shipping address.
* Fix - Hide the -OR- on Minicart when selecting "Enable hide button mode".
* Fix - Change Payment Method for a Subscription Paid with another Payment Method.
* Fix - Express Checkout on Product page not getting the values of the variations created by WooCommerce Product Add-Ons and Gravity Forms Product Add-ons
